### PR TITLE
add gitflow chapter

### LINKED ISSUE #33: Precipitate gitflow into the guide
Seems like we're moving to use gitflow [e.g. this link][gitflow] in a number of projects

Is this something we want to make official or only use for certain projects?

Also, seems wise to let users/contribs know in the README/any contrib docs that the project uses this approach so they can follow along accordingly (e.g. avoid PRs to main when almost always should be to dev). 

@tefirman @seankross 

[gitflow]: https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow

### PR DESCRIPTION
**Pull request by sckott**
fixes #33 

Please take a look. Seem good as a separate chapter, or should it live somewhere else? Look at in light and dark mode to make sure the diagram looks ok. Let me know if the diagram could be any better

### CODE REVIEW

File: gitflow.qmd [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+   merge dev
+   commit tag: "v2.0"
+```
+
+There are formal tools for using a Gitflow model, e.g., [nvie/gitflow](https://github.com/nvie/gitflow). However, we aren't using that tool or any others at the moment.
+
+There are components of Gitflow that we do not use. They are:
+
+- Release branches
+- Hotfix branches
```
Comment (seankross) [inline gitflow.qmd:43]: We do talk about using hotfix branches in the guide, and I like the concept, but I don't think we do too much that is "live" enough to require using them yet:

- https://deploy-preview-42--monumental-cassata-2f9bb7.netlify.app/codereview-dasl.html#high-level-approach:~:text=All%20Stable%20projects%20need%20branch%20protection%20on%20the%20main%20branch%20so%20that%20the%20main%20branch%20can%20only%20be%20updated%20via%20pull%20requests%20from%20dev%20or%20hotfix%2D%20branches%20with%20the%20approval%20of%20a%20project%20lead.
- https://deploy-preview-42--monumental-cassata-2f9bb7.netlify.app/maintenance.html#sec-versioning:~:text=Bug%20fixes%20and%20hotfixes%20cause%20the%20patch%20number%20to%20increment%3A%200.1.0%20%E2%86%92%200.1.1
- https://deploy-preview-42--monumental-cassata-2f9bb7.netlify.app/maintenance.html#branching:~:text=Developmental/incremental/fix/hotfix

File: gitflow.qmd [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## WILDS considerations
+
+- One thing we may consider doing in the future is making the `dev` branch the default branch in all WILDS repos, but that is not a rule right now
+- For WILDS repos following Gitflow, have a badge or something in the readme that gives info on Gitflow so contribs know? And/or put in CONTRIBUTING.md too?
```
Comment (seankross) [inline gitflow.qmd:48]: In CONTRIBUTING.md we should put some links to the most important parts of our docs.

File: gitflow.qmd [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+There are formal tools for using a Gitflow model, e.g., [nvie/gitflow](https://github.com/nvie/gitflow). However, we aren't using that tool or any others at the moment.
+
+There are components of Gitflow that we do not use. They are:
+
+- Release branches
+- Hotfix branches
+
+## WILDS considerations
+
+- One thing we may consider doing in the future is making the `dev` branch the default branch in all WILDS repos, but that is not a rule right now
```
[the remark this answers is not in the document]
Comment (sckott) [reply gitflow.qmd:47]: Yes, AFAIK, and I just tested it in a dummy repo - the default branch is the landing page of the repo too. i'll remove this sentence.

File: gitflow.qmd (hunk 2 above)
Comment (sckott) [reply gitflow.qmd:48]: For the `CONTRIBUTING.md` file are you referring to our getwilds repo wide file https://github.com/getwilds/.github/blob/main/CONTRIBUTING.md  ?

> most important parts of our docs

by docs do you mean this guide?

Comment (seankross) [reply gitflow.qmd:47]: Yes and yes.

Comment (sckott) [reply gitflow.qmd:47]: Okay, text added in the WILDS contrib file. https://github.com/getwilds/.github/commit/65a17f88679d3e562da14dcd5a3e4cd49cfd8fbb link to be added later when this PR is merged.

I think it makes sense then to remove the **WILDS considerations** section, yes?

**Comment by sckott** [comment]
@seankross @tefirman  Do we want to make gitflow something we use across all WILDS repos or only use for certain  projects? AFAICT we hadn't said we wanted to have a "rule" about this in our set of rules. This decision has a big impact on the wording in this chapter. Based on our past conversations, it seems like we want to have gitflow be optional such that each repo would need to indicate that they are using a gitflow/dev branch model (or let a contributor know if they accidentally submit a PR against `main`

File: gitflow.qmd (hunk 2 above)
Comment (seankross) [reply gitflow.qmd:47]: Sounds good and yes, if you could remove it that would be great. Thanks.

**Comment by seankross** [comment]
My feeling is that if a repo has a Stable badge, it has to use gitflow, otherwise it doesn't have to.

**Comment by sckott** [comment]
Sounds good to have gitflow be a rule for stable

**Comment by sckott** [comment]
So actually I don't think we need any further changes. we already have rules for this see https://deploy-preview-42--monumental-cassata-2f9bb7.netlify.app/maintenance 

Instead of just Stable we already said we want Prototype and Stable to have git flow essentially (main branch protection, only update main via PR from dev, hotfix branches)

**Comment by sckott** [comment]
Merging now

**Pull request merged**